[BUG] Typescript typecheck issue with generated axios code

##### Description

When generating code with generator type "typescript-axios" and release 7.19.0, there is a function generated in common.ts, which was introduced with release 7.19.0 (by pull request https://github.com/OpenAPITools/openapi-generator/pull/22642/files/5256bcb55fb76284bbedae0bccd1fe3de234bbac):

```
export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
    if (value instanceof Set) {
        return Array.from(value);
    } else {
        return value;
    }
}
```

The problem here is that the key parameter is not used, and typecheck now complains:

```
.../common.ts:99:61 - error TS6133: 'key' is declared but its value is never read.

99 export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
```

(Unfortunately it seems to be not possible to exclude generated files from typecheck)

##### openapi-generator version

7.19.0 (before: 7.18.0 where it worked)

--- a/modules/openapi-generator/src/main/resources/typescript-axios/common.mustache
+++ b/modules/openapi-generator/src/main/resources/typescript-axios/common.mustache
@@ -110,7 +110,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/echo_api/typescript-axios/build/common.ts
+++ b/samples/client/echo_api/typescript-axios/build/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts
+++ b/samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/composed-schemas/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/composed-schemas/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/default/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/default/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/es6-target/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/es6-target/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/test-petstore/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/test-petstore/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/with-complex-headers/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-complex-headers/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/with-fake-endpoints-models-for-testing-with-http-signature/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-fake-endpoints-models-for-testing-with-http-signature/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/with-interfaces-and-with-single-request-param/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-interfaces-and-with-single-request-param/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/with-interfaces/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-interfaces/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/with-node-imports/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-node-imports/common.ts
@@ -97,7 +97,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/with-npm-version-and-separate-models-and-api/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-npm-version-and-separate-models-and-api/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/with-npm-version/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-npm-version/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/with-single-request-parameters/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-single-request-parameters/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

--- a/samples/client/petstore/typescript-axios/builds/with-string-enums/common.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-string-enums/common.ts
@@ -96,7 +96,8 @@ export const setSearchParams = function (url: URL, ...objects: any[]) {
  * This function will run for every key-value pair encountered by JSON.stringify while traversing an object.
  * Converting a set to a string will return an empty object, so an intermediate conversion to an array is required.
  */
-export const replaceWithSerializableTypeIfNeeded = function(key: any, value: any) {
+// @ts-ignore
+export const replaceWithSerializableTypeIfNeeded = function(key: string, value: any) {
     if (value instanceof Set) {
         return Array.from(value);
     } else {

